Lib/json/tool.py

```diff
@@ -26,19 +26,26 @@ def main():
                         help='write the output of infile to outfile')
     parser.add_argument('--sort-keys', action='store_true', default=False,
                         help='sort the output of dictionaries alphabetically by key')
+    parser.add_argument('--jsonlines', action='store_true', default=False,
+                        help='parse input using the jsonlines format')
     options = parser.parse_args()
 
     infile = options.infile or sys.stdin
     outfile = options.outfile or sys.stdout
     sort_keys = options.sort_keys
+    jsonlines = options.jsonlines
     with infile:
         try:
-            obj = json.load(infile)
+            if jsonlines:
+                objs = [json.loads(line) for line in infile]
```

Comment:
Instead of loading all lines and only later dump all objects, it may be better to: load on object, dump it, then load the second object, etc.

The behavior difference is that first JSONs are written into outfile if they are valid, but if the last one is invalid. With the current code, it's all-or-nothing. Maybe it's done on purpose?

The other advantage of working on a single object is to reduce the memory footprint when working with large JSON files.

Reply:
Good idea.

Just make `objs` an iterator:
```python
objs = (json.loads(line) for line in infile)
```
or
```python
objs = map(json.loads, infile)
```

Reply:
Wow, that's great. Thank you for your guidance.